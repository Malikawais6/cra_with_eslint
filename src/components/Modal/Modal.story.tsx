import React from "react";
import { storiesOf } from "@storybook/react";

import { ModalComponent } from "./Modal";
import Line from "../Line/Line";
import { LegendSeriesData } from "../Line/__mock__/mockdata";
import { LineChartStyled } from "./Modal.style";

storiesOf("Modal", module).add("Default", () => (
  <ModalComponent title="Default Modal" visible={true}>
    <p>Content</p>
  </ModalComponent>
));

storiesOf("Modal", module).add(
  "Modal Without Footer",
  () => {
    return (
      <ModalComponent
        title="Gift Card Volume"
        visible={true}
        footer={false}
        width="960px"
      >
        <LineChartStyled>
          <Line
            showXAxis={true}
            showYAxis={true}
            showLegends={true}
            showLabels={true}
            showGridLines={true}
            showLineMarks={true}
            seriesData={LegendSeriesData}
          ></Line>
        </LineChartStyled>
      </ModalComponent>
    );
  },
  {
    info: { inline: true }
  }
);
