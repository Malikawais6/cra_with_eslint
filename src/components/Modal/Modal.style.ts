import styled from "styled-components";
import "antd/dist/antd.css";
import { Modal } from "antd";

export const ModalStyled = styled(Modal)`
  .ant-modal-body {
    padding-bottom: 41px;
    margin-top: -15px;
  }
  .ant-modal-close {
    padding-top: 10px;
    width: 64px;
    .ant-modal-close-x {
      font-size: 20px;
    }
  }
  .ant-modal-header {
    border-bottom: none;
    padding-top: 27px;
    .ant-modal-title {
      font-size: 24px;
      font-weight: 500;
    }
  }
`;
export const LineChartStyled = styled.div`
  height: 430px;
  width: 880px;
`;
