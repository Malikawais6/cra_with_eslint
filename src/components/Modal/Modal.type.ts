import { ReactNode } from "react";

export type Props = {
  title?: string;
  height?: string;
  width?: string;
  footer?: boolean;
  visible?: boolean;
  children?: ReactNode;
  handleOk?: () => void;
  handleCancel?: () => void;
};
