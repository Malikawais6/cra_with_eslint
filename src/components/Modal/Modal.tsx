import React from "react";
import { Modal as AntdModal } from "antd";
import "antd/dist/antd.css";

import { Props } from "./Modal.type";
import { ModalStyled } from "./Modal.style";

export const ModalComponent = (props: Props) => {
  return (
    <ModalStyled data-testid="modal-wrapper">
      <AntdModal
        title={props.title}
        visible={props.visible}
        onOk={props.handleOk}
        onCancel={props.handleCancel}
        footer={props.footer}
      >
        {props.children}
      </AntdModal>
    </ModalStyled>
  );
};
