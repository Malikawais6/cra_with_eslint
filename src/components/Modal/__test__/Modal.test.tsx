import * as React from "react";
import { render } from "@testing-library/react";

import { ModalComponent } from "../Modal";

function renderField() {
  return <ModalComponent visible={true} />;
}

it("Make sure modal renders properly", () => {
  const { getByTestId } = render(renderField());

  const ModalElement = getByTestId("modal-wrapper");

  expect(ModalElement.className.length).not.toBeLessThan(1);
});
