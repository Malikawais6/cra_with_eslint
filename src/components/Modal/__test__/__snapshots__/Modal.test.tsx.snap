// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Make sure modal renders properly 1`] = `
<div
  class="sc-bdVaJa bDWFJH"
/>
`;
